### ISSUE TITLE
GTK: allow scroll bar behaviours of applications such as Firefox and Thunderbird to be consistent with behaviours of other applications

### ISSUE THREAD

**Issue by grahamperrin** [trueos/trueos-core#227]
## tl;dr

In the RSS reader that is embedded in the desktop of Lumina, click in the scroll bar **anywhere far** below the bottom of the scroller. Observe the resulting movement of the scroller: down, exactly **one page** – that's good, expected. 

Now, for consistency: 

```
$ cat /usr/local/etc/gtk-3.0/settings.ini
[Settings]
# http://kb.mozillazine.org/Ui.scrollToClick
gtk-primary-button-warps-slider = false
$ cat ~/.config/gtk-3.0/settings.ini
[Settings]
# gtk-primary-button-warps-slider = true
$ 
```

## Essentially

At every installation of TrueOS Desktop, 
if `/usr/local/etc/gtk-3.0/settings.ini` does not exist 
then create that file with this content: 

```
[Settings]
# http://kb.mozillazine.org/Ui.scrollToClick
gtk-primary-button-warps-slider = false
```

## Ideally

Each user should have a prepared preference file that can override the system preference. 

If `~/.config/gtk-3.0/settings.ini` does not exist 
then create that file with this content: 

```
[Settings]
# gtk-primary-button-warps-slider = true
```

Removal of 
`# `
from the beginning of the commented-out line should allow the scroller in a subsequently launched GTK+ 3 application to move **farther** than one page in response to a single, **exactly-targeted** click in the scroll bar away from the scroller. 

## Application-specific overrides of both the system preference and the user preference

TrueOS Desktop need not offer a GUI to such things. For reference only: 

### Mozilla Firefox

At `about:config` create an integer preference named **ui.scrollToClick** with a value of either: 

- **0**, for traditional paginated scrolling; or 
- **1**, to force the more modern but contentious behaviour that involves exact targeting by the user. 

## Further reading

Including hints that some users may enjoy: 

[How to fix GTK3 scrollbar behavior](http://askubuntu.com/q/295988/25036) (2013-05-16)

[Scrolling in GTK+ | Goings on](https://blogs.gnome.org/mclasen/2013/08/05/scrolling-in-gtk/) (2013-08-05)

> … Clicking outside the thumb jumps to the position; paging is still available via right click …

[How To Change Scrollbar Action In GNOME 3.10](http://unix.stackexchange.com/q/182186/13260) (2015-01-31)

[Scroll bar erratic • KDE Community Forums](https://forum.kde.org/viewtopic.php?f=66&t=129512) (2015-11-26)

[Scrollbar moves to where I click](http://unix.stackexchange.com/q/276005/13260) (2016-04-12)

[Set Firefox scrollbar behavour to advance a single screen per click?](http://superuser.com/q/1071353/84988) (2016-04-29)

### Bugzilla@Mozilla: 

[803633 – [Linux gtk] Honor scrollbar preference of system](https://bugzilla.mozilla.org/show_bug.cgi?id=803633)

Last but not least, [1260067 – Scrollbars broken on FreeBSD?](https://bugzilla.mozilla.org/show_bug.cgi?id=1260067) with a [cross-referencing comment](https://bugzilla.mozilla.org/show_bug.cgi?id=1260067#c8) from me.

**Comment by pkgdemon** [comment]
We cannot make a change like this without shipping overlays.  Which we have largely removed when removing pc-extractoverlay.  It would be best to pursue these changes you are after upstream if they are things which should be set globally for all new users.
